ios/Classes/core/PMPlugin.m

```diff
@@ -38,15 +38,29 @@ - (void)onMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
   PMManager *manager = self.manager;
 
   if ([call.method isEqualToString:@"requestPermission"]) {
-    [PHPhotoLibrary requestAuthorization:^(PHAuthorizationStatus status) {
-        BOOL auth = PHAuthorizationStatusAuthorized == status;
-        [manager setAuth:auth];
-        if (auth) {
-          [handler reply:@1];
-        } else {
-          [handler reply:@0];
-        }
-    }];
+      if (@available(iOS 14, *)) {
```

Comment:
```suggestion
#ifdef  __IPHONE_14_0
```

Reply:
Wouldn't this exclude iPad builds?

Reply:
Macro is used in precompiling. This is just to determine whether a macro is defined in your SDK, not if at runtime.
Therefore, when you use xcode14, your IOS SDK contains this macro, but 13.0 does not. It does not conflict with your target. You can try to run it under xcode14's iPad target.